src/main/scala/com/cloudera/sparkts/models/ARIMAX.scala

```diff
@@ -0,0 +1,583 @@
+import com.cloudera.sparkts.MatrixUtil.{dvBreezeToSpark, mBreezeToSpark, toBreeze}
+import com.cloudera.sparkts.UnivariateTimeSeries.{differencesOfOrderD, inverseDifferencesOfOrderD}
+import org.apache.commons.math3.analysis.{MultivariateFunction, MultivariateVectorFunction}
+import org.apache.commons.math3.optim.{InitialGuess, MaxEval, MaxIter, SimpleValueChecker}
+import org.apache.commons.math3.optim.nonlinear.scalar.{GoalType, ObjectiveFunction, ObjectiveFunctionGradient}
+import org.apache.commons.math3.optim.nonlinear.scalar.gradient.NonLinearConjugateGradientOptimizer
+import org.apache.spark.mllib.linalg.{DenseVector, Vector}
+
+/**
+  * The autoregressive integrated moving average with explanatory variables (ARIMAX)
+  * Y_{t} = Beta*X_{t} + ARIMA_MODEL
+  * Link [[http://support.sas.com/resources/papers/proceedings16/11823-2016.pdf]]
+  */
+
+object ARIMAX {
+
+  /**
+    * Given a time series, fit a non-seasonal ARIMAX model of order (p, d, q), where p represents
+    * the autoregression terms, d represents the order of differencing, q moving average error
+    * terms, xMaxLag the maximum lag order for exogenous variables. If includeOriginalXreg is true, the
+    * model is fitted with the non-lagged exogenous variables. If includeIntercept is true, the model is
+    * fitted with an intercept. In order to select the appropriate order of the model, users are advised
+    * to inspect ACF and PACF plots, or compare the values of the objective function.
+    *
+    * @param p Autoregressive order
+    * @param d Differencing order
+    * @param q Aoving average order
+    * @param ts Time series to which to fit an ARIMAX(p, d, q) model
+    * @param xreg A matrix of exogenous variables
+    * @param xregMaxLag The maximum lag order for the dependent variable
+    * @param includeOriginalXreg A boolean flag indicating if the non-lagged exogenous variables should
+    *                            be included. Default is true.
+    * @param includeIntercept If true the model is fit with an intercept term. Default is true
+    * @param userInitParams A set of user provided initial parameters for optimization. If null
+    *                       (default), initialized using Hannan-Rissanen algorithm. If provided,
+    *                       order of parameter should be: intercept term, AR parameters (in
+    *                       increasing order of lag), MA parameters (in increasing order of lag) and
+    *                       coefficients for exogenous variables (xregMaxLag + 1) for each column.
+    */
+  def fitModel(p: Int,
+               d: Int,
+               q: Int,
+               ts: Vector,
+               xreg: Matrix[Double],
+               xregMaxLag: Int,
+               includeOriginalXreg: Boolean = true,
+               includeIntercept: Boolean = true,
+               userInitParams: Array[Double] = null): ARIMAXModel = {
```

Comment:
A common pattern in scala is to use the Option[T] datatype to specify absence of value, rather than null.